docs/content/sips/00x-cloudevent-trigger.md

```diff
@@ -0,0 +1,263 @@
+title = "SIP xxx - CloudEvents trigger"
+template = "main"
+date = "2022-04-25T14:53:30Z"
+---
+
+Summary: A CloudEvents trigger for spin.
+
+Owner: jiazho@microsoft.com
+
+Created: April 24, 2022
+
+Updated: April 24, 2022
+
+## Background
+
+Currently spin supports two triggers, one for Redis messages and one for HTTP requests. [CloudEvents](https://cloudevents.io/) are a new standard for eventing and received huge interests from the major cloud providers. Supporting CloudEvents could make spin a great solution for writing serverless applications. 
+
+
+## Proposal
+
+This document proposes adding features in the spin SDK to support CloudEvents. CloudEvents itself is a envelop for the underlying transport protocol, such as AMQP, Kafka, HTTP, etc. This proposal aims at providing a CloudEvent component for the [HTTP Protocol Bindings](https://github.com/cloudevents/spec/blob/main/cloudevents/bindings/http-protocol-binding.md). Here is an example shows the mapping of an event with an HTTP POST request in CloudEvent's binary format.
+```
+POST /someresource HTTP/1.1
+Host: webhook.example.com
+ce-specversion: 1.0
+ce-type: com.example.someevent
+ce-time: 2018-04-05T03:56:24Z
+ce-id: 1234-1234-1234
+ce-source: /mycontext/subcontext
+    .... further attributes ...
+Content-Type: application/json; charset=utf-8
+Content-Length: nnnn
+{
+    ... application data ...
+}
+```
+
+Creating an HTTP CloudEvents trigger is done by defining the top level application trigger in spin. The following code snippet shows the definition of a HTTP CloudEvents trigger.
+```toml
+# spin.toml
+trigger = { type = "http", base = "/", schema = "cloudevents" }
+```
+
+The added `schema` attribute in trigger will tell spin application that it will expect the HTTP request and responses are CloudEvents.
+
+> Note that the `schema` attribute is not the same as the `schema` attribute in the CloudEvents spec. The `schema` attribute in the spec is the schema of the payload.
+
+We also allow users to define individual component to be CloudEvents components. For example, we could define a HTTP CloudEvents component in spin.toml:
+
+```toml
+# spin.toml
+trigger = { type = "http", base = "/" }
+
+[[component]]
+id = "hello"
+source = "target/wasm32-wasi/release/spinhelloworld.wasm"
+description = "A simple component that returns hello."
+schema = "cloudevents"
```

Comment:
Are both `schema` fields needed then?
Could this be controlled by the component configuration only?

Also,  this would have to be part of `component.trigger` instead.